app/src/main/java/ch/epfl/sdp/connectivity/ConcreteConnectivityBroker.java

```diff
@@ -33,23 +35,58 @@
     private LocationManager locationManager;
     private Context context;
 
+    private boolean checkInternetConnection() {
+        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo nInfo = cm.getActiveNetworkInfo();
+
+        boolean connectionAvailable = nInfo != null && nInfo.isAvailable() && nInfo.isConnected();
+
+        try {
+            InetAddress ipAddr = InetAddress.getByName("google.com");
+            //You can replace it with your name
+            return connectionAvailable && !ipAddr.equals("");
+
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
     private ConnectivityManager.NetworkCallback internetNetworkCallback = new ConnectivityManager.NetworkCallback() {
+
         @Override
         public void onCapabilitiesChanged(@NonNull Network network, @NonNull NetworkCapabilities networkCapabilities) {
             super.onCapabilitiesChanged(network, networkCapabilities);
-            if (networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)) {
+            if (checkInternetConnection()) {
+                Log.e("BROKER", "Internet capabilities changed");
```

Comment:
Should we keep this log?

Reply:
It was useful to check which one of the two methods was called